### PR TITLE
PR: Add some color schemes from Eclipse

### PR DESCRIPTION
**Pull request by PanderMusubi**
## Description of Changes

See issue

### Issue(s) Resolved

Fixes https://github.com/spyder-ide/spyder/issues/8371

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: PanderMusubi

### CODE REVIEW

**Comment by ccordoba12** [comment]
Thanks for your contribution!

@CAM-Gerlach, could you review this one? Thanks!

**Comment by PanderMusubi** [comment]
@CAM-Gerlach there will be one more commit, please review it only when that is in.

**Comment by PanderMusubi** [comment]
@CAM-Gerlach latest commit is there. :)

**Comment by CAM-Gerlach** [comment]
@CAM-Gerlach @ccordoba12 Okay, thanks—looking at it now.

File: spyder/config/main.py [hunk 1]
```diff
@@ -661,7 +667,425 @@
...
+              'black/pastel/normal':     ('#c0c0c0', False, False),
+              'black/pastel/keyword':    ('#82677e', False, False),
+              'black/pastel/builtin':    ('#82677e', False, False),
+              'black/pastel/definition': ('#82677e', False, False),
+              'black/pastel/comment':    ('#7d8c93', False, False),
+              'black/pastel/string':     ('#c78d9b', False, False),
+              'black/pastel/number':     ('#c78d9b', False, False),
+              'black/pastel/instance':   ('#678cb1', False, False),
+              # ---- frontenddev (Eclipse color theme) ----
+              'frontenddev/name':        "frontenddev",
```
Comment (CAM-Gerlach) [inline spyder/config/main.py]: Title Case.

File: spyder/config/main.py [hunk 2]
```diff
@@ -661,7 +667,425 @@
...
+              'inkpot/normal':     ('#cfbfad', False, False),
+              'inkpot/keyword':    ('#808bed', False, False),
+              'inkpot/builtin':    ('#87cefa', False, False),
+              'inkpot/definition': ('#87cefa', False, False),
+              'inkpot/comment':    ('#cd8b00', False, False),
+              'inkpot/string':     ('#ffcd8b', False, False),
+              'inkpot/number':     ('#ffcd8b', False, False),
+              'inkpot/instance':   ('#cfbfad', False, False),
+              # ---- minimal (Eclipse color theme) ----
+              'minimal/name':        "minimal",
```
Comment (CAM-Gerlach) [inline spyder/config/main.py]: Title Case.

**Comment by CAM-Gerlach** [comment]
Thanks @PanderMusubi ; this pretty much looks good to me now. You're going to need to ``rebase`` on the latest ``master``, since we also just updated the Spyder Dark color scheme which resulted in a conflict with yours.

While you're at it, could you ``fixup`` (via e.g, interactive rebase) your second and fourth commits into your first and third, and edit the latter's commit message to conform to the universal imperative tense and capatilized convention ("Add color schemes converted from Eclipse color themes", "Optimize and add current cell colors to new Eclipse color schemes")? Thanks.

**Comment by CAM-Gerlach** [comment]
Okay, that's going to be a lot of screenshots, heh. I can do it a little later.

**Comment by PanderMusubi** [comment]
For this and future PRs, please use https://github.com/PanderMusubi/convert-theme-to-scheme which will generate the Python code to include. It also generated an INI file to easily test the schemes. I will leave the squashing and rebasing to you. If you need me to help with convert-theme-to-scheme, please let me know.

**Comment by goanpeca** [comment]
I will update the PR and post the screenshots

**Comment by CAM-Gerlach** [comment]
Thanks @goanpeca ! I had started going through them, but with the very long lag to switch syntax schemes on my machine it ended up getting bumped for other priorities.

**Comment by steff456** [comment]
These are the screenshots for the default themes

**Emacs**

image

----
**Idle**

image

----
**Monokai**

image
----
**Pydev**

image
----
**Scintilla**
image

----
**Solarized Dark**
image

----
**Solarized Light**
image

----
**Spyder**
image

----
**Spyder Dark**
image

----
**Zenburn**
image

**Comment by ccordoba12** [comment]
@steff456, those are the current schemes. Please post screenshots of these ones: frontenddev, Havenjark, Gedit Original Oblivion, etc (please see the changes to config/main.py for all the new themes).

**Comment by steff456** [comment]
**Black Pastel**

image
----
**Frontenddev**

image
----
**Gedit Original Oblivion**

image
----
**Havenkark**

image

----
**Inkpot**

image

----
**Minimal**

image

----
**Mr**

image

----
NightLionAptana Theme
image

----
**Notepad++ Like**

image

----
**Oblivion**

image

----
** Obsidian**
image

----
**Pastel**
image

----
**RecognEyes**
image

----
**Retta**

image

----
**Roboticket**

image

----
**Schuss**
image

----
**Sublime Text Monokai Extended**
image

----
**Sublime Text 2**
image

----
**Sunburst**
image

----
**Tango**
image

----
**Vibrant Ink**
image

----
**Wombat**
image

**Comment by goanpeca** [comment]
@ccordoba12 we could add these styles as plugins?

It might also make sense to not keep the styles on the `spyder.ini` anymore.

**Comment by ccordoba12** [comment]
> we could add these styles as a plugin that adds them?

Not right now.

> It might also make sense to not keep the styles on the spyder.ini anymore.

Good point @goanpeca!

@steff456, please move the entire `appearance` section as a dict to a new file called `spyder/config/appearance.py` with the name `APPEARANCE`, and import it from there to `spyder/config/main.py`.

**Comment by ccordoba12** [comment]
I'll take a look this weekend at what schemes should stay and be removed.

@goanpeca, any particular preferences?

**Comment by goanpeca** [comment]
I like the dark ones :-) pastel

File: spyder/config/appearance.py [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+import sys
+
+from spyder.config.fonts import MEDIUM, MONOSPACE, SANS_SERIF, SMALL
+
+WIN = os.name == 'nt'
+LINUX = sys.platform.startswith('linux')
+
+# Default themes
+APPEARANCE = {
+              'icon_theme': 'spyder 3',
```
Comment (ccordoba12) [inline spyder/config/appearance.py]: Please only leave four spaces of indentation between the left margin of the file and this line, and align all the rest of the lines on this dict accordingly.

File: spyder/config/appearance.py [hunk 4]
```diff
@@ -0,0 +1,641 @@
+import os
```
Comment (ccordoba12) [inline spyder/config/appearance.py]: This is missing the header we add to all files in our source code. Please look at any other file and copy the first ten lines or so, which contain our copyright and the `coding` statement.

Comment (ccordoba12) [reply spyder/config/appearance.py]: And please also add a docstring to this module, below those comments and above this line.

**Comment by ccordoba12** [comment]
Ok, I think we should remove these themes:

* Black pastel (too low contrast)
* Frontenddev (too similar to our Emacs theme)
* Gedit Original Oblivion (too low contrast)
* Havenkark (too low contrast and I don't like the select line color)
* Mr (too similar to IDLE)
* RecognEyes (too similar to our theme)
* Schuss (too low contrast)
* Sublime Text 2 (I don't like the colors and we already have Monokai)
* Sunburst (too low contrast)
* Tango (too low contrast)
* Wombat (I don't like the colors)

----

@spyder-ide/core-developers, @spyder-ide/junior-developers, what do you think?

**Comment by ccordoba12** [comment]
@steff456, please rename `Notepad++ Like` to `Notepad++`.

**Comment by ccordoba12** [comment]
@steff456, I just noticed that none of the new themes provides a `currentcell` color. Please add one by using a color picker and getting a lighter version of the background color for the dark themes. For the light themes you can use the light yellow used in the Spyder theme.

**Comment by goanpeca** [comment]
@ccordoba12 to prevent this in the future I suggest we add a simple test that will iterate and check that:
- All the expected keys are found for every theme
- All the values are correct color values (or True or False)

Thoughts?

**Comment by ccordoba12** [comment]
They key was there, the problem was that the color was not defined to something useful.

**Comment by goanpeca** [comment]
Ah ok got it

**Comment by ccordoba12** [comment]
@steff456, last comments about the themes we chose:

* Notepad++: `currentline` cannot be distinguished from `currentcell`.
* Pastel: The difference  between `currentline` and `currentcell` is very small. Please make the difference clearer.
* Roboticket: `currentline` cannot be distinguished from `currentcell`.

File: spyder/config/appearance.py [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+    'font/italic': False,
+    'font/bold': False,
+    'rich_font/family': SANS_SERIF,
+    'rich_font/size': SMALL if (LINUX or WIN) else MEDIUM,
+    'rich_font/italic': False,
+    'rich_font/bold': False,
+    'ui_theme': 'automatic',
+    'names': ['emacs', 'idle', 'monokai', 'pydev', 'scintilla',
+              'spyder', 'spyder/dark', 'zenburn', 'solarized/light',
+              'solarized/dark', 'inkpot', 'minimal', 'nightlion/aptana/theme',
```
Comment (ccordoba12) [inline spyder/config/appearance.py]: Let's rename `nightlion/aptana/theme` to `nightlion`.

File: spyder/config/appearance.py [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+              'notepad++', 'oblivion', 'obsidian', 'pastel', 'retta',
+              'roboticket', 'sublime/te/mo/ex', 'vibrant/ink'],
```
Comment (ccordoba12) [inline spyder/config/appearance.py]: Let's rename `sublime/te/mo/ex` to `sublime-monokai/extended` and  `vibrant/ink` to `vibrant-ink`.

File: spyder/config/main.py [hunk 7]
```diff
@@ -1,729 +1,523 @@
...
-except Exception:
-    CONF = UserConfig('spyder', defaults=DEFAULTS, load=False,
-                      version=CONF_VERSION, subfolder=SUBFOLDER, backup=True,
-                      raw_mode=True)
-
-# Removing old .spyder.ini location:
-old_location = osp.join(get_home_dir(), '.spyder.ini')
-if osp.isfile(old_location):
-    os.remove(old_location)
+# -*- coding: utf-8 -*-
```
Comment (ccordoba12) [inline spyder/config/main.py]: @steff456, please revert the changes made by the original author to the line endings of this file, so we can get a good diff about what really changed in this PR (the current line endings are LF, but the previous ones were CRLF).

File: spyder/config/main.py [hunk 8]
```diff
@@ -1,729 +1,523 @@
...
+# Config instance
+# =============================================================================
+# IMPORTANT NOTES:
+# 1. If you want to *change* the default value of a current option, you need to
+#    do a MINOR update in config version, e.g. from 3.0.0 to 3.1.0
+# 2. If you want to *remove* options that are no longer needed in our codebase,
+#    or if you want to *rename* options, then you need to do a MAJOR update in
+#    version, e.g. from 3.0.0 to 4.0.0
+# 3. You don't need to touch this value if you're just adding a new option
+CONF_VERSION = '50.1.0'
```
Comment (ccordoba12) [inline spyder/config/main.py]: Please bump this value to `50.2.0` so that the new themes are visible in our preferences by previous users.
